### PR TITLE
Feature/unit test patient dao

### PR DESCRIPTION
**Pull request by rashmi-bambhania**
Test cases for database layer and Code Refactoring

### CODE REVIEW

File: app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/add/FetchTestRecordFragment.kt [hunk 1]
```diff
@@ -80,9 +80,11 @@ class FetchTestRecordFragment : Fragment(R.layout.fragment_fetch_covid_test_resu
...
 
                     if (state.errorData != null) {
-                        requireContext().showError(
-                            getString(state.errorData.title),
-                            getString(state.errorData.message)
+                        AppAlertDialog.showConfirmationAlertDialog(
+                            context = requireContext(),
+                            title = getString(state.errorData.title),
+                            msg = getString(state.errorData.message),
+                            positiveBtnMsg = getString(android.R.string.ok)
```
Comment (pinakinkansara) [inline app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/add/FetchTestRecordFragment.kt:87]: do not use system string resources please declare our own resource in string.xml

**Review by pinakinkansara — CHANGES_REQUESTED** [review]

File: data/src/androidTest/java/ca/bc/gov/data/local/dao/PatientDaoTest.kt [hunk 2]
```diff
@@ -1,49 +1,105 @@
...
+        Assert.assertNotEquals(1, result)
+    }
+
+    @Test
+    fun checkPatientId() = runBlocking {
+        // Given
+        val patient = getPatient1()
+
+        // When
+        patientDao.insertPatient(patient)
```
[the remark this answers is not in the document]
Comment (rashmi-bambhania) [reply data/src/androidTest/java/ca/bc/gov/data/local/dao/PatientDaoTest.kt:58]: Done

File: data/src/androidTest/java/ca/bc/gov/data/local/dao/PatientDaoTest.kt [hunk 3]
```diff
@@ -1,49 +1,105 @@
...
 
-        val patient =
-            PatientEntity(firstName = "Pinakin", lastName = "kansara", dateOfBirth = Instant.now())
-        val id = patientDao.insertPatient(patient)
+        // When
+        val result = patientDao.insertPatient(patient)
 
-        assertTrue(id > 0)
+        // Then
+        Assert.assertEquals(1, result)
```
[the remark this answers is not in the document]
Comment (rashmi-bambhania) [reply data/src/androidTest/java/ca/bc/gov/data/local/dao/PatientDaoTest.kt]: Done

File: data/src/androidTest/java/ca/bc/gov/data/local/BaseDataBaseTest.kt [hunk 4]
```diff
@@ -43,12 +49,72 @@ abstract class BaseDataBaseTest {
         tearDown()
     }
 
-    protected fun getPatient() =
+    protected fun getPatient1() =
```
[the remark this answers is not in the document]
Comment (rashmi-bambhania) [reply data/src/androidTest/java/ca/bc/gov/data/local/BaseDataBaseTest.kt:52]: Patient is an DB entity. It is not required to use mockito to mock Data classes.

File: app/src/main/java/ca/bc/gov/bchealth/utils/AppAlertDialog.kt [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            .setMessage(msg)
+            .setPositiveButton(positiveBtnMsg) { dialog, _ ->
+                positiveBtnCallback?.invoke()
+                dialog.dismiss()
+            }
+            .setNegativeButton(negativeBtnMsg) { dialog, _ ->
+                negativeBtnCallback?.invoke()
+                dialog.dismiss()
+            }
+            .setCancelable(false)
```
[the remark this answers is not in the document]
Comment (rashmi-bambhania) [reply app/src/main/java/ca/bc/gov/bchealth/utils/AppAlertDialog.kt]: Done

File: app/src/main/java/ca/bc/gov/bchealth/ui/travelpass/FetchFederalTravelPassFragment.kt [hunk 6]
```diff
@@ -84,9 +84,11 @@ class FetchFederalTravelPassFragment : Fragment(R.layout.fragment_fetch_travel_p
...
 
                     if (uiState.errorData != null) {
-                        requireContext().showError(
-                            getString(uiState.errorData.title),
-                            getString(uiState.errorData.message)
+                        AppAlertDialog.showConfirmationAlertDialog(
+                            context = requireContext(),
+                            title = getString(uiState.errorData.title),
+                            msg = getString(uiState.errorData.message),
+                            positiveBtnMsg = getString(android.R.string.ok)
```
[the remark this answers is not in the document]
Comment (rashmi-bambhania) [reply app/src/main/java/ca/bc/gov/bchealth/ui/travelpass/FetchFederalTravelPassFragment.kt]: Done

File: app/src/main/java/ca/bc/gov/bchealth/utils/AppAlertDialog.kt [hunk 7]
```diff
@@ -0,0 +1,40 @@
+package ca.bc.gov.bchealth.utils
+
+import android.content.Context
+import androidx.appcompat.app.AlertDialog
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+
+object AppAlertDialog {
```
[the remark this answers is not in the document]
Comment (rashmi-bambhania) [reply app/src/main/java/ca/bc/gov/bchealth/utils/AppAlertDialog.kt]: Done

File: app/src/main/java/ca/bc/gov/bchealth/utils/AppAlertDialog.kt [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    private lateinit var dialogAlert: AlertDialog
+
+    private fun isDialogShowing(): Boolean =
+        this::dialogAlert.isInitialized && dialogAlert.isShowing
+
+    fun showConfirmationAlertDialog(
```
[the remark this answers is not in the document]
Comment (rashmi-bambhania) [reply app/src/main/java/ca/bc/gov/bchealth/utils/AppAlertDialog.kt]: Done

File: app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/add/FetchTestRecordFragment.kt (hunk 1 above)
Comment (rashmi-bambhania) [reply app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/add/FetchTestRecordFragment.kt]: Done

**Review by pinakinkansara — APPROVED** [review]

**Pull request merged**